collector/src/main/java/com/bitmovin/analytics/BitmovinAnalytics.java

```diff
@@ -221,7 +221,9 @@ public void onError(ErrorCode errorCode) {
 
         data.setErrorCode(errorCode.getErrorCode());
         data.setErrorMessage(errorCode.getDescription());
-        data.setErrorData(serialize(errorCode.getErrorData()));
```

Comment:
Would it make sense to change this in the `serialize` method? If an object is `null`, we never want it to be serialized to the string I think? Not sure in which places we use that method, but I can't think of a reason where we want to send `"null"` to the server, and if it's really intended, I think it would make more sense to implement the opposite case: `val serialized = (obj == null) ? "null": serialize(obj)`.

Reply:
Well i would expect this method to return "null" for null because it should just serialize it
what about adding a new method `serializeOrNull(Any): String?` which is clear on the return type?

Reply:
So far I have only seen this instance where we explicitly serialise an object.
Regarding your suggestion:
What about `customDataX` fields? We usually don't set them to safe storage.

Reply:
TBH I would not expect a method to serialize `null` to a `"null"` string. I don't hink if you are the consumer of the serialized string, you'd want to start parsing a `"null"` string, instead you'd expect the object to be set to `null` already.
@Tigraine thoughts?

Reply:
I also don't think sending a string `"null"` to the server makes any sense TBH.
If the `errorDetails` are null then they should be null. This is better then addressed at the presentation layer.

Reply:
But where would you but the check? I'd expect the serialize method to not do anything for null (only return null), instead of having another method that wraps it

Reply:
Resolved by adding the null check into the serialisation method